src/wallet/wallet.cpp

```diff
@@ -2057,11 +2055,6 @@ OutputType CWallet::TransactionChangeType(const std::optional<OutputType>& chang
         // As of 2021 about 80% of all SH are wrapping WPKH, so use that
         return OutputType::P2SH_SEGWIT;
     }
-    const bool has_legacy_spkman(GetScriptPubKeyMan(OutputType::LEGACY, /*internal=*/true));
-    if (has_legacy_spkman && any_pkh) {
-        // Currently pkh is the only type supported by the LEGACY spkman
-        return OutputType::LEGACY;
-    }
 
     if (has_bech32m_spkman) {
         return OutputType::BECH32M;
```

Comment:
I do like the idea of nudging people to use bech32m by default, but what's the point of a "default" if we then default to something else?